libFMS_mod/libFMS.F90

```diff
@@ -0,0 +1,471 @@
+                     monotonic_array, string_array_index
+
+  !> horiz_interp
+  use horiz_interp_mod, only: horiz_interp, horiz_interp_new, horiz_interp_del, &
+                              horiz_interp_init, horiz_interp_end
+  use horiz_interp_type_mod, only: horiz_interp_type, assignment(=), CONSERVE, &
+                              BILINEAR, SPHERICA, BICUBIC, stats
+  !! used via horiz_interp
+  ! horiz_interp_bicubic_mod, horiz_interp_bilinear_mod
+  ! horiz_interp_conserve_mod, horiz_interp_spherical_mod
+
+  !> interpolator
+  use interpolator_mod, only: interpolator_init, interpolator, interpolate_type_eq, &
+                              obtain_interpolator_time_slices, unset_interpolator_time_flag, &
+                              interpolator_end, init_clim_diag, query_interpolator, &
+                              interpolate_type, CONSTANT, &
+                              INTERP_WEIGHTED_P, INTERP_LINEAR_P, INTERP_LOG_P, &
+                              ZERO_INTERP=>ZERO, & !! conflicts with mpp_parameter's ZERO 
+                              read_data_interp=>read_data !! conflicts with fms2_io interface
+
+  !> memutils
+  use memutils_mod, only: memutils_init, print_memuse_stats
+
+  !> monin_obukhov
+  use monin_obukhov_mod, only: monin_obukhov_init, monin_obukhov_end, &
+                               mo_drag, mo_profile, mo_diff, stable_mix
+  use monin_obukhov_inter, only: monin_obukhov_diff, monin_obukhov_drag_1d, &
+                               monin_obukhov_solve_zeta, monin_obukhov_derivative_t, &
+                               monin_obukhov_derivative_m, monin_obukhov_profile_1d, &
+                               monin_obukhov_integral_m, monin_obukhov_integral_tq, &
+                               monin_obukhov_stable_mix
+
+  !> mosaic
+  use mosaic2_mod, only: get_mosaic_ntiles, get_mosaic_ncontacts, &
+                      get_mosaic_grid_sizes, get_mosaic_contact, &
+                      get_mosaic_xgrid_size, get_mosaic_xgrid, &
+                      calc_mosaic_grid_area, calc_mosaic_grid_great_circle_area, &
+                      is_inside_polygon, &
+                      get_mosaic_tile_grid_2 => get_mosaic_tile_grid !overloaded in fms2_io
+  use grid_mod, only: get_grid_ntiles, get_grid_size, get_grid_cell_centers, &
+                      get_grid_cell_vertices, get_grid_cell_Area, get_grid_comp_area, &
+                      define_cube_mosaic
+  use gradient_mod, only: gradient_cubic, calc_cubic_grid_info
+
+  !> mpp
+  use mpp_mod, only: mpp_init_test_full_init, mpp_init_test_init_true_only, &
+                     mpp_init_test_peset_allocated, mpp_init_test_clocks_init, &
+                     mpp_init_test_datatype_list_init, mpp_init_test_logfile_init, &
+                     mpp_init_test_read_namelist, mpp_init_test_etc_unit, &
+                     mpp_init_test_requests_allocated, stdin, stdout, stderr, &
```

Comment:
I don't think we want to expose mpp_init_test_* to the public.  Our unit tests can grab them directly from mpp_mod.

Reply:
Removed these in 056a29f